### PR TITLE
feat: add translate function

### LINKED ISSUE #13432: Feature: add translate() funtion
**Summary**

to reduce the nested calls of replace().
eg.
use
translate('abcde','abc','ABC') instead of
replace(replace(replace('abcde','a','A'),'b','B'),'c','C')
ref https://docs.oracle.com/cd/B13789_01/server.101/b10759/functions181.htm
https://www.postgresql.org/docs/current/functions-string.html

Syntax:
```
TRANSLATE( <subject>, <sourceAlphabet>, <targetAlphabet> )
```

Example:
```
SELECT TRANSLATE('abcde','abc','ABC');
+--------------------------------+
| TRANSLATE('abcde','abc','ABC') |
|--------------------------------|
| ABCdef                         |
+--------------------------------+
```

### PR DESCRIPTION
**Pull request by guojidan**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #13432 issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13535)

### CODE REVIEW

**Comment by guojidan** [comment]
@TCeason please review this pr : )

File: src/query/functions/src/scalars/string.rs [hunk 1]
```diff
@@ -239,6 +239,30 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+        "translate",
+        |_, _, _, _| FunctionDomain::Full,
+        vectorize_with_builder_3_arg::<StringType, StringType, StringType, StringType>(
+            |str, from, to, output, _| {
+            if from.is_empty() || from == to {
+                output.put_slice(str);
+                output.commit_row();
+                return;
+            }
+            let to_len = to.len();
```
Comment (BohuTANG) [inline src/query/functions/src/scalars/string.rs:252]: Should check: `from.len() == to.len()`?

**Comment by TCeason** [comment]
This pr adds a string function translate. Please review this pr. @andylokandy

**Review by sundy-li — APPROVED** [review]

File: src/query/functions/src/scalars/string.rs (hunk 1 above)
Comment (guojidan) [reply src/query/functions/src/scalars/string.rs:252]: The length of from and to can be different

Replaces each character in string that matches a character in the from set with the corresponding character in the to set. If from is longer than to, occurrences of the extra characters in from are deleted.

**Comment by BohuTANG** [comment]
When merge with main, there are some mismatched types error: https://github.com/datafuselabs/databend/actions/runs/6806765528/job/18508525426?pr=13535#step:4:7590

**Review by andylokandy — APPROVED** [review]

**Comment by BohuTANG** [comment]
Thanks for the contribution.

**Pull request merged**